envoy/server/filter_config.h

```diff
@@ -154,6 +154,13 @@ class NamedUpstreamNetworkFilterConfigFactory : public ProtocolOptionsFactory {
                                                                 CommonFactoryContext& context) PURE;
 
   std::string category() const override { return "envoy.filters.upstream_network"; }
+
+  /**
+   * @return bool true if this filter must be the last filter in a filter chain, false otherwise.
+   */
+  virtual bool isTerminalFilterByProto(const Protobuf::Message&, ServerFactoryContext&) {
```

Comment:
Why do we need parameters here? Is there a factory that can produce both terminal and non-terminal filters?

Reply:
Oh, I see it's just copied from above.